crates/utils/Cargo.toml

```diff
@@ -13,42 +13,112 @@ name = "lemmy_utils"
 path = "src/lib.rs"
 doctest = false
 
+[[bin]]
+name = "lemmy_util_bin"
+path = "src/main.rs"
+required-features = ["default"]
+
 [lints]
 workspace = true
 
 [features]
-full = ["ts-rs"]
+default = [
+  "error",
+  "request",
+  "response",
+  "settings",
+  "misc",
+  "rate-limit",
+  "apub",
+  "cache-header",
+  "email",
+]
+full = ["default", "ts-rs"]
+ts-rs = ["dep:ts-rs"]
+error-type = ["dep:serde", "dep:strum"]
+error = [
+  "error-type",
+  "dep:serde_json",
+  "dep:anyhow",
+  "dep:tracing-error",
+  "dep:diesel",
+  "dep:http",
+  "dep:actix-web",
+]
+request = ["dep:reqwest-middleware", "dep:tracing", "dep:strum"]
+response = ["error", "dep:actix-web"]
+settings = [
+  "error",
+  "dep:deser-hjson",
+  "dep:regex",
+  "dep:urlencoding",
+  "dep:doku",
+  "dep:url",
+  "dep:once_cell",
+  "dep:smart-default",
+]
+rate-limit = [
+  "error",
+  "dep:enum-map",
+  "dep:tracing",
+  "dep:actix-web",
+  "dep:futures",
+  "dep:tokio",
+  "dep:once_cell",
+]
+apub = ["dep:openssl"]
+cache-header = ["dep:actix-web"]
+email = [
+  "error",
+  "settings",
+  "dep:html2text",
+  "dep:lettre",
+  "dep:uuid",
+  "dep:rosetta-i18n",
+]
+misc = [
+  "settings",
+  "dep:itertools",
+  "dep:markdown-it",
+  "dep:tokio",
+  "dep:futures",
+  "dep:tracing",
+]
```

Comment:
I dont see the purpose of all these distinct features. It should be like the other crates, with a single feature `full`.

Reply:
Add `#[cfg(feature = "full")]` to more things

Reply:
@dullbananas I tried that at first, but I couldn't get it to work. The main issue is that originally, the only dependency that wasn't used by default was ts-rs. Everything else, including the parts that depend on diesel and actix, were included in the default feature. I'll see if I can reduce the number of features, but I don't think I'll be able to keep it like it was before where there's just one dependency omitted from default.

Reply:
Have you checked how other Lemmy crates handle the `full` feature? It should work in exactly the same way here.

Reply:
I've taken a look at some of the other crates. What's throwing me off is that in the case of other crates, the only 2 options are default and full. The utils crate, on the other hand, needs to support 3 cases:

- Just the error type and its direct dependencies
- Everything except for `ts-rs` (default)
- Everything including `ts-rs` (full)

If I want to keep it to just `default` and` full`, I would have to make `default` only export the error types and `full` would be used to use all of the features that are in the current default while also including the `ts-rs`.

Reply:
Seems like the most flexible then would be having `ts-rs` and `error` as the only features, both default off.

Reply:
`error` should be fine to have included in `default`. What I need to do is be able to use just `error` without anything else in `default`. This might let me simplify things, though.